### ISSUE TITLE
Ambiguity problem in the detection of the ArUco marker when it is parallel to the camera plane

### ISSUE THREAD

**Issue by DanielBilbao12** [opencv/opencv#21672]
##### System information (version)

- OpenCV => 4.5.5
- Operating System / Platform => Windows 10, 64 Bit
- Compiler/IDE=> Pycharm 2021.3.2
- Language => Python 3.9

##### Detailed description

I am developing a vision system based on ArUco visual markers. The ArUco library used is "cv2.aruco.DICT_4x4_50" and the marker I have used is 2 cm on each side.

I have made the calibration using a 3.6 cm board for each square, obtaining results that I consider satisfactory.

The problem appeared when I started to validate the results provided by the vision system. To perform the validation, a UR10e collaborative robot has been used, in which the ArUco marker has been strategically placed on the tool connected to it. In this way, the robot tool is positioned in a known spatial position and, using a Logitech C525 camera, the marker is located and its position and orientation with respect to the camera is calculated (actually the position of the tool is calculated, for this the reference system is transferred from the center of the aruco to the center of the tool using homogeneous transformation matrices).

The problem I have is that when the tool and its marker are placed just in front of the camera, i.e. without any rotation in X or Y, I seem to have an ambiguity problem because the vision system returns a rotation value in Y, as you can see in the following example:

[X,Y,Z,Rx,Ry,Rz] Real= [0mm, 0mm, 250mm, 0deg, 0deg, 0deg, 20deg]

[X,Y,Z,Rx,Ry,Rz] of the vision system= [-0.26mm, 0.58mm, 254.2mm, 0.64deg, -7.9deg, 19.23deg]

_**Example Images Here**_
![1](https://user-images.githubusercontent.com/66834161/156344175-67019ab0-1bad-46a4-a443-f6fb24e12c67.PNG)
![1_Res](https://user-images.githubusercontent.com/66834161/156344182-bd5f4ba7-6dc2-48f4-9c9d-b9fd1ba2dcd2.PNG)

At first I assumed it was an offset error introduced when defining the camera reference system, but when I rotate the Y, the result I get is correct, as in the following example:

[X,Y,Z,Rx,Ry,Rz] Real= [0mm, 0mm, 250mm, 0deg, 30deg, 20deg]

[X,Y,Z,Rx,Ry,Rz] of the vision system= [-0.088mm, 0.31mm, 255.64mm, 2.05deg, 28.3deg, 19.45deg]

_**Example Images Here**_
![2](https://user-images.githubusercontent.com/66834161/156344247-84dfe286-24b4-42d1-b2dc-a127d8a05401.PNG)
![2_Res](https://user-images.githubusercontent.com/66834161/156344252-de600bd6-04ca-4a97-b381-f6e21bc91e44.PNG)

In short, how can I solve the ambiguity problem for cases where my ArUco is in front of the camera, without any X or Y rotation? Is it really an ambiguity problem for using a single small marker or is there some other problem?

##### Notes

- Used resolution -> 1280x720
- AutoFocus and Auto White Balance -> disabled
- CornerRefinementMethod ->1
- I have read the OpenCV documentation but I did not found any detailed solution to this issue.

If you need more detailed information, please do not hesitate to ask for it.
